### PR TITLE
feat: change JsonResponse::DEFAULT_ENCODING_OPTIONS with real value

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Serializer/JsonEncoder.php [hunk 1]
```diff
@@ -34,7 +34,8 @@ public function __construct(string $format, BaseJsonEncoder $jsonEncoder = null)
 
         // Encode <, >, ', &, and " for RFC4627-compliant JSON, which may also be embedded into HTML.
         $this->jsonEncoder = $jsonEncoder ?: new BaseJsonEncoder(
-            new JsonEncode(JsonResponse::DEFAULT_ENCODING_OPTIONS), new JsonDecode(true)
+        // The value "15" will be replaced by  `JsonResponse::DEFAULT_ENCODING_OPTIONS` when Symfony HttpFoundation 2.7 support will be dropped
```
Comment (dunglas) [inline src/Serializer/JsonEncoder.php]: We don't use Markdown in comments (backticks)

File: src/Serializer/JsonEncoder.php [hunk 2]
```diff
@@ -34,7 +34,8 @@ public function __construct(string $format, BaseJsonEncoder $jsonEncoder = null)
 
         // Encode <, >, ', &, and " for RFC4627-compliant JSON, which may also be embedded into HTML.
         $this->jsonEncoder = $jsonEncoder ?: new BaseJsonEncoder(
-            new JsonEncode(JsonResponse::DEFAULT_ENCODING_OPTIONS), new JsonDecode(true)
+        // The value "15" will be replaced by  "JsonResponse::DEFAULT_ENCODING_OPTIONS" when Symfony HttpFoundation 2.7 support will be dropped
```
Comment (teohhanhui) [inline src/Serializer/JsonEncoder.php]: Why not use the PHP constants directly? `JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_AMP | JSON_HEX_QUOT`

We do not necessarily want to stick to `JsonResponse::DEFAULT_ENCODING_OPTIONS`. For example, I think `JSON_UNESCAPED_UNICODE` should be the default in this day and age. And we can remove one unused import too...

File: src/Serializer/JsonEncoder.php [hunk 3]
```diff
@@ -34,7 +34,8 @@ public function __construct(string $format, BaseJsonEncoder $jsonEncoder = null)
 
         // Encode <, >, ', &, and " for RFC4627-compliant JSON, which may also be embedded into HTML.
```
Comment (teohhanhui) [inline src/Serializer/JsonEncoder.php]: This comment is outdated. See https://github.com/symfony/http-foundation/blob/99b9cc2044fe1d36f4c141da1496190cf5591ddf/JsonResponse.php#L30

File: src/Serializer/JsonEncoder.php (hunk 2 above)
Comment (Simperfit) [inline src/Serializer/JsonEncoder.php]: I agree, this was just a workaround but if we could use the php directly why not ? @dunglas WDYT ?

Comment (dunglas) [inline src/Serializer/JsonEncoder.php]: Yes we must do that it's cleaner.

**Pull request merged**